grpclog/internal/loggerv2_test.go

```diff
@@ -51,13 +126,243 @@ func TestLoggerV2Severity(t *testing.T) {
 	}
 }
 
-// check if b is in the format of:
-//
-//	2017/04/07 14:55:42 WARNING: WARNING
-func checkLogForSeverity(s int, b []byte) error {
-	expected := regexp.MustCompile(fmt.Sprintf(`^[0-9]{4}/[0-9]{2}/[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2} %s: %s\n$`, severityName[s], severityName[s]))
-	if m := expected.Match(b); !m {
-		return fmt.Errorf("got: %v, want string in format of: %v", string(b), severityName[s]+": 2016/10/05 17:09:26 "+severityName[s])
+func TestLoggerV2PrintFuncDiscardOnlyInfo(t *testing.T) {
+	buffers := []*bytes.Buffer{nil, new(bytes.Buffer), new(bytes.Buffer)}
+	logger := NewLoggerV2(io.Discard, buffers[warningLog], buffers[errorLog], LoggerV2Config{})
+	loggerTp := logger.(*loggerT)
+
+	loggerTp.printFunc(infoLog, makeLogFuncErr(t))
```

Comment:
probably a comment stating that `makeLogFuncErr` won't be called since infoLog is io.Discard?

Reply:
Also, based on above NewLoggerV2(..) logic, since warningLog is not io.discard, infoLog shouldn't be either?

Reply:
> probably a comment stating that makeLogFuncErr won't be called since infoLog is io.Discard?

Added

Reply:
> Also, based on above NewLoggerV2(..) logic, since warningLog is not io.discard, infoLog shouldn't be either?

I'm not entirely sure what you mean?

If you mean the multi-writer logic:

```
	if warningW != io.Discard || infoW != io.Discard {
		warnLogW = io.MultiWriter(infoW, warningW)
	}
``` 

That's only setting `io.MultiWriter` on higher level logs. That is, `warn` is a MultiWriter for warn+info, but `info` is still a `io.Discard` (which is what this test is verifying).

If `logger.print(warnLog` is called, it will print to both `info` and `warn` (that behaviour predates this PR; this PR doesn't change any behaviour). So, when `info` is `io.Discard` and `warn` isn't, `fn` is called once, and the result written to the `io.MultiWriter` which will write to warn and info (which happens to be `io.Discard`). The `Printf` is only called once via `fn`, which is inevitable. (That behaviour is verified in `TestLoggerV2PrintFuncAllCombinations`.)

Does that answer your question?

Reply:
Ah ok. In that case a docstring for test will be helpful. Something like this. 

```
TestLoggerV2PrintFuncDiscardOnlyInfo ensures that logs at the INFO level are discarded when set to io.Discard, while logs at other levels (WARN, ERROR) are still printed. It does this by using a custom error function that raises an error if the logger attempts to print at the INFO level, ensuring early return when io.Discard is used.
```

Reply:
make sure to keep the comments cols upto 80 chars

Reply:
Added